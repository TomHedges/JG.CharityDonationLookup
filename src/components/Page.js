import React, { Component } from "react";
import { hot } from "react-hot-loader";
import XRemainder from "./XRemainder";
import HeaderContent from "./HeaderContent";
import FooterContent from "./FooterContent";

class Page extends Component {
  render() {
    const HEADING = "JustGiving Donation Feed";
    return (
      <>
        <HeaderContent heading={HEADING} />
        <XRemainder />
        <FooterContent heading={HEADING} />
      </>
    );
  }
}

export default hot(module)(Page);
