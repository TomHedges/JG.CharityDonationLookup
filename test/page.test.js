import Page from "../src/components/Page";

/**
 * Unit tests for the Page component
 */

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Page />);
});

describe("<Page /> rendering", () => {
  it("Should render <Page /> component", () => {
    wrapper = shallow(<Page />);
    expect(wrapper.exists()).toBe(true);
  });
});

describe("<Page /> displays HeaderContent, MainContent and FooterContent", () => {
  it("Should display a single <HeaderContent>", () => {
    expect(wrapper.find("HeaderContent")).toHaveLength(1);
  });

  it("Should display a single <MainContent>", () => {
    expect(wrapper.find("MainContent")).toHaveLength(1);
  });

  it("Should display a single <FooterContent>", () => {
    expect(wrapper.find("FooterContent")).toHaveLength(1);
  });

  it("Snapshot contains HeaderContent, MainContent and FooterContent components", () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
