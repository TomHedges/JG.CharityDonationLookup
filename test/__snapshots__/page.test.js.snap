// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> displays HeaderContent, MainContent and FooterContent Snapshot contains HeaderContent, MainContent and FooterContent components 1`] = `
Array [
  <div
    className="header_content side_pad"
  >
    <img
      className="jg_logo"
      src={[Function]}
    />
    <h1>
      JustGiving Donation Feed
    </h1>
  </div>,
  <div
    className="main_content side_pad"
  >
    <div
      className="charity_selection"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          htmlFor="charity_id"
        >
          Please enter Charity ID and select charity from dropdown list:
        </label>
        <div
          className="dropdown_wrapper"
        >
          <input
            className="charity_id_selector"
            id="charity_id"
            onChange={[Function]}
            type="text"
            value=""
          />
          <div
            className="dropdown_content_shown"
          />
        </div>
      </form>
    </div>
  </div>,
  <div
    className="footer_content side_pad"
  >
    <p>
      Â© Tom Hedges 2019 - 
      JustGiving Donation Feed
    </p>
  </div>,
]
`;
